lib/TikZImage.pm

```diff
@@ -156,60 +157,87 @@ sub draw {
 	my $working_dir = WeBWorK::PG::ImageGenerator::makeTempDirectory(WeBWorK::PG::IO::ww_tmp_dir(), "tikz");
 	my $data;
 
+	my $ext = $self->ext;
+	my $svgMethod = $self->svgMethod;
+	my $latex     = WeBWorK::PG::IO::externalCommand('latex');
+	my $pdflatex  = WeBWorK::PG::IO::externalCommand('pdflatex');
+
 	my $fh;
-	open($fh, ">", "$working_dir/image.tex")
-		or warn "Can't open $working_dir/image.tex for writing.";
-	chmod(0777, "$working_dir/image.tex");
-	print $fh $self->header;
-	print $fh $self->tex =~ s/\\\\/\\/gr . "\n";
-	print $fh $self->footer;
-	close $fh;
 
-	my $ext = $self->ext;
-	my $tex_ext = $ext eq 'svg' && $self->svgMethod eq 'dvisvgm' ? 'dvi' : 'pdf';
-	my $latex_binary = WeBWorK::PG::IO::externalCommand($tex_ext eq 'dvi' ? 'latex' : 'pdflatex');
-
-	# Generate the pdf file.
-	system "cd " . $working_dir . " && $latex_binary image.tex > pdflatex.stdout 2> /dev/null";
-
-	if (-r "$working_dir/image.$tex_ext") {
-		chmod(0777, "$working_dir/image.$tex_ext");
-		# Convert the file to the appropriate type of image file
-		if ($ext eq 'svg') {
-			if ($self->svgMethod eq 'dvisvgm') {
-				system WeBWorK::PG::IO::externalCommand('dvisvgm') .
-					" $working_dir/image.dvi --no-fonts --output=$working_dir/image.svg > /dev/null 2>&1";
-			} else {
-				system WeBWorK::PG::IO::externalCommand($self->svgMethod) .
-					" $working_dir/image.pdf $working_dir/image.svg > /dev/null 2>&1";
-			}
-		} elsif ($ext ne 'pdf') {
-			system WeBWorK::PG::IO::externalCommand('convert') .
-				join('',map {" -$_ " . $self->convertOptions->{input}->{$_}} (keys %{$self->convertOptions->{input}})) .
-				" $working_dir/image.pdf" .
-				join('',map {" -$_ " . $self->convertOptions->{output}->{$_}} (keys %{$self->convertOptions->{output}})) .
-				" $working_dir/image.$ext > /dev/null 2>&1";
+	# Create either one or two tex files with one small difference:
+	# set pgfsysdriver to pgfsys-dvisvgm.def for a tex file that dvisvgm will use
+	# Then make only the dvi, only the pdf, or both in case we are making tgz with svg via dvisvgm
+	if (($ext eq 'svg' || $ext eq 'tgz') && $svgMethod eq 'dvisvgm') {
+		open($fh, ">", "$working_dir/image-dvisvgm.tex")
+			or warn "Can't open $working_dir/image-dvisvgm.tex for writing.";
+		my @header = $self->header;
+		splice @header, 1, 0, "\\def\\pgfsysdriver{pgfsys-dvisvgm.def}\n";
+		print $fh @header;
+		print $fh $self->tex =~ s/\\\\/\\/gr . "\n";
+		print $fh $self->footer;
+		close $fh;
+		system "cd $working_dir && $latex image-dvisvgm.tex > latex.stdout 2> /dev/null && mv image-dvisvgm.dvi image.dvi";
```

Comment:
This should probably use the `mv` external program via `WeBWorK::PG::IO::externalCommand('mv')`.